converge/settings.py

```diff
@@ -98,14 +98,28 @@ def clone_git_repo(git_url, settings_dir, git_subdir=None):
         run_command(cp_cmd)
 
 
-def main():
+def get_rc_config():
     rc_config_default = {'APP_MODE': 'dev',
                          'SETTINGS_DIR': None,
                          'GIT_SETTINGS_REPO': None,
                          'GIT_SETTINGS_SUBDIR': None}
-
     rc_config = parse_osenv(rc_config_default)
     rc_config = parse_rc(rc_config_default)
+    return rc_config
+
+
+def detect_mode():
+    config = get_rc_config()
```

Comment:
@rayandas I think we don't need to parse config every time. Also we can create `settings.APP_MODE` property instead of `detect_mode` method.
cc: @shon

Reply:
Every time when `settings.detect_mode()` will be called, `get_rc_config()` will check current APP_MODE and return the mode.(prod/dev/stage/beta) And if it is in supported_app_mode then it will return the mode. 
And `settings.APP_MODE` property as in property function?

Reply:
@rayandas `settings.APP_MODE` should return current app mode

Reply:
@gauravr okay so the function name sill be APP_MODE.
I'll make some changes after finding a way to call `get_rc_config` only once and push the chnges.

Reply:
@rayandas don't need to create `detect_mode` function, when the first time we parse app mode, create `APP_MODE` at that time only.